cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -672,6 +860,40 @@ protected Future<ReconcileResult<ClusterRoleBinding>> connectInitClusterRoleBind
     @SuppressWarnings({ "rawtypes" })
     abstract Future<AutoRestartStatus> previousAutoRestartStatus(Reconciliation reconciliation, String connectorName, CustomResource resource);
 
+    // Methods for working with connector offsets
+
+    /**
+     * Returns the value of strimzi.io/connector-offsets annotation on the provided custom resource.
+     *
+     * @param resource          Custom resource (a KafkaConnector or KafkaMirrorMaker2) instance to check
+     * @param connectorName     Name of the connector being reconciled
+     *
+     * @return  The value of the strimzi.io/connector-offsets annotation on the provided custom resource.
+     */
+    @SuppressWarnings({ "rawtypes" })
+    abstract KafkaConnectorOffsetsAnnotation getConnectorOffsetsAnnotation(CustomResource resource, String connectorName);
+
+    /**
+     * Patches the custom resource to remove the connector-offsets annotation
+     *
+     * @param reconciliation    Reconciliation marker
+     * @param resource          Custom resource from which the annotation should be removed
+     *
+     * @return  Future that indicates the operation completion
+     */
+    @SuppressWarnings({ "rawtypes" })
+    abstract Future<Void> removeConnectorOffsetsAnnotation(Reconciliation reconciliation, CustomResource resource);
+
+    /**
+     * Returns the key to use for either writing connector offsets to a ConfigMap or fetching connector offsets
+     * from a ConfigMap.
+     *
+     * @param connectorName Name of the connector that is being managed.
+     *
+     * @return The String to use when interacting with ConfigMap resources.
+     */
+    abstract String getConnectorOffsetsConfigMapEntryKey(String connectorName);
```

Comment:
The names with `Annotation` in singular sound strange if you then actually use multiple of them. Maybe we can try to find some more neutral names?

Reply:
What about `getConnectorOffsetsAnnotation` -> `getConnectorOffsetsOperation` or `getConnectorOffsetsAction`?
And `removeConnectorOffsetsAnnotation` -> `removeConnectorOffsetsAnnotations?

Reply:
We could even rename the enum `KafkaConnectorOffsetsAnnotation` to `KafkaConnectorOffsetsOperation`

Reply:
Yeah, I think calling it operation or action would make it a bit more clear. Not sure I would change the `KafkaConnectorOffsetsAnnotation` name as that is the enum with the possible values, or? So that makes sense to and with `Annotation`.